### PR TITLE
mesa: Added 24.2.5

### PR DESCRIPTION
**Pull request by kenmays**
> **Mesa 24.2.5 for Haiku R1 - x86, x64, riscv64 platforms**

Ref: https://docs.mesa3d.org/relnotes/24.2.5.html

Note (built on Haiku R1b5 x86): 
> Compiled with libglvnd 1.7.0 to package completion!
> Vulkan 1.3.289 API
> New shader cache implementation, reducing filesystem overhead.

@kallisti5
@X547

### CODE REVIEW

**Comment by waddlesplash** [comment]
CC @X547

File: sys-libs/mesa/mesa-24.2.0.recipe [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	haiku${secondaryArchSuffix}_devel
+	libglvnd${secondaryArchSuffix}_devel
+	devel:libz$secondaryArchSuffix
+	devel:libLLVM_17$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libvulkan$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	python3.10$secondaryArchSuffix
+	mako_python310
```
Comment (X547) [inline sys-libs/mesa/mesa-24.2.0.recipe]: `pyyaml_python310` is missing.

**Review by X547 — CHANGES_REQUESTED** [review]

File: sys-libs/mesa/mesa-24.2.0.recipe (hunk 1 above)
Comment (kenmays) [reply sys-libs/mesa/mesa-24.2.0.recipe:80]: Fixef

**Comment by X547** [comment]
What is stopping from merging this?

File: sys-libs/mesa/mesa-24.2.1.recipe [hunk 2]
```diff
@@ -146,7 +146,7 @@ INSTALL()
 	# Configure Mesa EGL Vendor Renderer
 	cp "$portDir/additional-files/50_mesa.json" "$addOnsDir/opengl/egl_vendor.d/"
 	# Configure vulkan lavapipe render (Fix path)
-	sed -i "s/\/packages\/mesa-.*\/.self\/lib\/libvulkan_lvp.so/.\/libvulkan_lvp.so/" $addOnsDir/vulkan/icd.d/lvp_icd.x86_64.json
+#	sed -i "s/\/packages\/mesa-.*\/.self\/lib\/libvulkan_lvp.so/.\/libvulkan_lvp.so/" $addOnsDir/vulkan/icd.d/lvp_icd.x86_64.json
```
Comment (waddlesplash) [inline sys-libs/mesa/mesa-24.2.1.recipe:149]: What's going on here?

**Comment by kenmays** [comment]
@x547 - Closing. Can migrate once DRM ioctl is reworked for R1.

**Pull request closed without merging**